src/main.rs

```diff
@@ -1187,31 +1143,102 @@ fn parse_and_output_man_pages(
     unimplemented!();
 }
 
-fn parsers_to_try() -> Vec<Box<dyn ManParser>> {
-    vec![
-        Box::new(Type1),
-        Box::new(Type2),
-        Box::new(Type3),
-        Box::new(Type4),
-        Box::new(TypeDarwin),
-        Box::new(TypeDeroff),
-    ]
+macro_rules! mantypes {
+    ($($typ: tt),*) => {
+        $(
+        impl From<$typ> for ManType {
+            fn from(t: $typ) -> Self {
+                ManType::$typ(t)
+            }
+        }
+        )*
+
+        #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+        enum ManType {
+            $($typ($typ),)*
+        }
+
+        impl ManType {
+            const ALL: &'static [ManType] = &[$(ManType::$typ($typ),)*];
```

Comment:
Just to verify that I read this correctly: This creates a single static array of all the available ManType's, on the `ManType` struct/"namespace"?

Reply:
Yeap. Replaces the previous vector explicitly constructed by `parsers_to_try`.